lib/seattleflu/id3c/cli/command/etl/redcap_det_childcare.py

```diff
@@ -0,0 +1,554 @@
+
+            specimen_observation_entry = create_specimen_observation_entry(
+                specimen_reference = specimen_reference,
+                patient_reference = patient_reference,
+                encounter_reference = encounter_reference)
+        else:
+            LOG.info('Creating encounter for record instance without sample')
+
+        enrollment_questionnaire_entry = None
+        encounter_questionnaire_entry = None
+        operational_questionnaire_entry = None
+
+        if event_type == EventType.ENROLLMENT:
+            enrollment_questionnaire_entry = create_enrollment_questionnaire_response(
+            enrollment, study_arm, patient_reference, encounter_reference)
+
+        # The SECONDARY arm gets "encounter" surveys in the ENROLLMENT event.
+        if event_type == EventType.ENCOUNTER or study_arm == StudyArm.SECONDARY:
+            encounter_questionnaire_entry = create_encounter_questionnaire_response(
+            redcap_record_instance, patient_reference, encounter_reference)
+
+            operational_questionnaire_entry = create_operational_questionnaire_response(
+            redcap_record_instance, patient_reference, encounter_reference)
+
+        current_instance_entries = [
+            encounter_entry,
+            enrollment_questionnaire_entry,
+            encounter_questionnaire_entry,
+            operational_questionnaire_entry,
+            specimen_entry,
+            specimen_observation_entry,
+        ]
+
+        persisted_resource_entries.extend(list(filter(None, current_instance_entries)))
+
+
+    return create_bundle_resource(
+        bundle_id = str(uuid4()),
+        timestamp = datetime.now().astimezone().isoformat(),
+        source = f'{REDCAP_URL}{enrollment.project.id}/{enrollment.id}',
+        entries = list(filter(None, persisted_resource_entries))
+    )
+
+
+def get_encounter_date(record: REDCapRecord, event_type: EventType) -> Optional[str]:
+    """
+    Determine the encounter_date from the given REDCap *record*.
+    """
+    encounter_date = None
+
+    if event_type == EventType.ENROLLMENT:
+        encounter_date = record.get('enrollment_date')
+
+    elif event_type == EventType.ENCOUNTER:
+        if record.get('today_2'): #Symptom Check
+            encounter_date = record['today_2']
+        elif record.get('kit_reg_date'): #Swab Kit Reg
+            encounter_date = record['kit_reg_date']
+        elif record.get('samp_process_date'): #Post Collection Data Entry Qc
+            encounter_date = record['samp_process_date'].split()[0]
+
+    return encounter_date
+
+
+def create_encounter_id(record: REDCapRecord) -> str:
+    """
+    Create the hashed encounter_id from the REDCap *record*.
+    """
+    if record.event_name:
+        redcap_event_name = record.event_name
+    else:
+        redcap_event_name = ''
+
+    if record.repeat_instance:
+        redcap_repeat_instance = str(record.repeat_instance)
+    else:
+        redcap_repeat_instance = ''
```

Comment:
No harm in keeping this, but the Childcare REDCap project does not have repeat instances.

Reply:
No need to leave extraneous code in there that could confuse us (me) three months from now. Removed.
I made the change in the "FIXUP: jover's feedback" commit.